### PR TITLE
Fixes #183 start time end time optional

### LINKED ISSUE #183: StartTime and EndTime should be optional in simulation control (DDI plot) as well DDI Simulation

### PR DESCRIPTION
**Pull request by msevestre**

### CODE REVIEW

File: schemas/OSP_Qualification_Plan_Schema.json [hunk 1]
```diff
@@ -41,6 +41,54 @@
...
+        "EndTime": {
+          "type": "number"
+        },
+        "TimeUnit": {
+          "$ref": "#/definitions/timeUnit"
+        }
+      },
+      "anyOf": [
+        {
+          "required": [
```
Comment (msevestre) [inline schemas/OSP_Qualification_Plan_Schema.json:65]: @Yuri05 I did not manage to make TimeUnit a must when EndTime is entered (or StartTime). Do you know how we can do that?

Comment (Yuri05) [reply schemas/OSP_Qualification_Plan_Schema.json:65]: theoretically should be possible with the new keywords from json schema draft 2019 or 2020
https://json-schema.org/understanding-json-schema/reference/conditionals.html#dependentrequired

E.g. something like this:
```
                          "SimulationControl": {
                            "type": "object",
                            "properties": {
                              "Project": {
                                "$ref": "#/definitions/nonEmptyString"
                              },
                              "Simulation": {
                                "$ref": "#/definitions/nonEmptyString"
                              },
                              "StartTime": {
                                "type": "number"
                              },
                              "EndTime": {
                                "type": "number"
                              },
                              "TimeUnit": {
                                "$ref": "#/definitions/timeUnit"
                              }
                            },
                            "required": [
                              "Project",
                              "Simulation"
                            ],
                            "dependentRequired": {
                              "StartTime": [
                                "TimeUnit"
                              ],
                              "EndTime": [
                                "TimeUnit"
                              ]
                            },
                            "additionalProperties": false
                          },
```

For this, we must switch from schema draft 7 to schema draft 2020-12
`"$schema": "https://json-schema.org/draft/2020-12/schema",`

But for some reason I cannot use it: VSCode complains
grafik

You could try if it works for you.

Comment (Yuri05) [reply schemas/OSP_Qualification_Plan_Schema.json:65]: ok, it was just the proxy issue, now I can use the latest draft schema.

So I tried 2 different options: `dependentRequired `for SimulationControl and `dependentSchemas `for SimulationDDI - but somehow both did not work. I was able e.g. to define Start/EndTime without the time unit.
[OSP_Qualification_Plan_Schema.zip](https://github.com/Open-Systems-Pharmacology/QualificationPlan/files/7142798/OSP_Qualification_Plan_Schema.zip)
So theoretically this should be the way to go, but something is wrong.

```
                          "SimulationControl": {
                            "type": "object",
                            "properties": {
                              "Project": {
                                "$ref": "#/definitions/nonEmptyString"
                              },
                              "Simulation": {
                                "$ref": "#/definitions/nonEmptyString"
                              },
                              "StartTime": {
                                "type": "number"
                              },
                              "EndTime": {
                                "type": "number"
                              },
                              "TimeUnit": {
                                "$ref": "#/definitions/timeUnit"
                              }
                            },
                            "required": [
                              "Project",
                              "Simulation"
                            ],
                            "dependentRequired": {
                              "StartTime": [
                                "TimeUnit"
                              ],
                              "EndTime": [
                                "TimeUnit"
                              ]
                            },
                            "additionalProperties": false
                          },
                          "SimulationDDI": {
                            "type": "object",
                            "properties": {
                              "Project": {
                                "$ref": "#/definitions/nonEmptyString"
                              },
                              "Simulation": {
                                "$ref": "#/definitions/nonEmptyString"
                              },
                              "StartTime": {
                                "type": "number"
                              },
                              "EndTime": {
                                "type": "number"
                              },
                              "TimeUnit": {
                                "$ref": "#/definitions/timeUnit"
                              }
                            },
                            "required": [
                              "Project",
                              "Simulation"
                            ],
                            "dependentSchemas": {
                              "StartTime": {
                                "properties": {
                                  "TimeUnit": {
                                    "$ref": "#/definitions/timeUnit"
                                  }
                                },
                                "required": [
                                  "TimeUnit"
                                ]
                              },
                              "EndTime": {
                                "properties": {
                                  "TimeUnit": {
                                    "$ref": "#/definitions/timeUnit"
                                  }
                                },
                                "required": [
                                  "TimeUnit"
                                ]
                              }
                            },
                            "additionalProperties": false
                          }
```

Comment (Yuri05) [reply schemas/OSP_Qualification_Plan_Schema.json:65]: So I

Comment (Yuri05) [reply schemas/OSP_Qualification_Plan_Schema.json:65]: ok, finally managed to do it (with the currently used json schema draft 7, so no need to switch to the newer version).
[OSP_Qualification_Plan_Schema.zip](https://github.com/Open-Systems-Pharmacology/QualificationPlan/files/7143284/OSP_Qualification_Plan_Schema.zip)


```
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "definitions": {
   ...
    "SimulationWithOptionalStartAndEndTime": {
      "type": "object",
      "properties": {
        "Project": {
          "$ref": "#/definitions/nonEmptyString"
        },
        "Simulation": {
          "$ref": "#/definitions/nonEmptyString"
        },
        "StartTime": {
          "type": "number"
        },
        "EndTime": {
          "type": "number"
        },
        "TimeUnit": {
          "$ref": "#/definitions/timeUnit"
        }
      },
      "required": [
        "Project",
        "Simulation"
      ],
      "dependencies": {
        "StartTime": {
          "required": [
            "TimeUnit"
          ]
        },
        "EndTime": {
          "required": [
            "TimeUnit"
          ]
        },
        "TimeUnit": {
          "anyOf": [
            {
              "required": [
                "StartTime"
              ]
            },
            {
              "required": [
                "EndTime"
              ]
            }
          ]
        }
      },
      "additionalProperties": false
    }
  },
...
                          "SimulationControl": {
                            "$ref": "#/definitions/SimulationWithOptionalStartAndEndTime"
                          },
                          "SimulationDDI": {
                            "$ref": "#/definitions/SimulationWithOptionalStartAndEndTime"
                          }
...

```

**Comment by msevestre** [comment]
wow... Crazy... well done

**Pull request merged**